demo_nodes_py/demo_nodes_py/parameters/set_parameters_callback.py

```diff
@@ -0,0 +1,85 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from rcl_interfaces.msg import SetParametersResult
+import rclpy
+from rclpy.node import Node
+from rclpy.parameter import Parameter
+
+
+# Example usage: changing param1 successfully will result in setting of param2.
+# ros2 service call /set_parameters_callback/set_parameters rcl_interfaces/srv/SetParameters
+#       "{parameters: [{name: "param1", value: {type: 3, double_value: 1.0}}]}"
+
+# node for demonstrating correct usage of pre_set, on_set
+# and post_set parameter callbacks
+class SetParametersCallback(Node):
+    
+    def __init__(self):
+        super().__init__('set_parameters_callback')
+        self.declare_parameter('param1', 0.0)
+        self.declare_parameter('param2', 0.0)
+
+        # tracks 'param1' value
+        self.internal_tracked_param_1 = self.get_parameter('param1').value
+
+        # tracks 'param2' value
+        self.internal_tracked_param_2 = self.get_parameter('param2').value
+
+        # setting another parameter from the callback is possible
+        # we expect the callback to be called for param2
+        def preSetParametersCallback(parameter_list):
+            modified_parameters = parameter_list.copy()
+            for param in parameter_list:
+                if param.name == "param1":
+                    modified_parameters.append(Parameter('param2', Parameter.Type.DOUBLE, 4.0))
+
+            return modified_parameters
+
+        # validation callback
+        def onSetParametersCallback(parameter_list):
+            result = SetParametersResult()
+            for param in parameter_list:
+                if param.name == 'param1':
+                    result.successful = True
+                    result.reason = 'success param1'
+                elif param.name == 'param2':
+                    result.successful = True
+                    result.reason = 'success param2'
+
+            return result
+
+        # can change internally tracked class attributes
+        def postSetParametersCallback(parameter_list):
+            for param in parameter_list:
+                if param.name == 'param1':
+                    self.internal_tracked_param_1 = param.value
+                elif param.name == 'param2':
+                    self.internal_tracked_param_2 = param.value
+
+        self.add_pre_set_parameters_callback(preSetParametersCallback)
+        self.add_on_set_parameters_callback(onSetParametersCallback)
+        self.add_post_set_parameters_callback(postSetParametersCallback)
+
+
+def main(args=None):
+    rclpy.init(args=args)
+    node = SetParametersCallback()
+    rclpy.spin(node)
+    node.destroy_node()
+    rclpy.shutdown()
```

Comment:
similar to our other demos, I think we should wrap this in a try-except-finally to avoid noisy stack traces during shutdown, e.g.

https://github.com/ros2/demos/blob/abcfe53757d5100bcb5b4785f5909c760538bc6c/demo_nodes_py/demo_nodes_py/topics/talker.py#L44-L50

Reply:
I don't have the access rights to push, what should I do? I guess under normal circumstances I would create a fork and submit a PR but since the PR is already opened not sure. Thanks!